source/IntegrationTests/Behaviours/IncomingRequests/GivenWholesaleServicesRequestV2WithDelegationTests.cs

```diff
@@ -0,0 +1,1064 @@
+        // It is very important that the generated data is correct,
+        // since (almost) all assertion after this point is based on this data
+        var requestCalculatedWholesaleServicesInputV1 = message!.ParseInput<RequestCalculatedWholesaleServicesInputV1>();
+        var requestCalculatedWholesaleServicesAccepted = WholesaleServicesResponseEventBuilder
+            .GenerateAcceptedFrom(requestCalculatedWholesaleServicesInputV1);
+
+        await GivenWholesaleServicesRequestAcceptedIsReceived(requestCalculatedWholesaleServicesAccepted);
+
+        // Act
+        var originalActorPeekResults = await WhenActorPeeksAllMessages(
+            originalActor.ActorNumber,
+            originalActor.ActorRole,
+            peekDocumentFormat);
+
+        var delegatedActorPeekResults = await WhenActorPeeksAllMessages(
+            delegatedToActor.ActorNumber,
+            delegatedToActor.ActorRole,
+            peekDocumentFormat);
+
+        // Assert
+        PeekResultDto peekResult;
+        using (new AssertionScope())
+        {
+            originalActorPeekResults.Should().BeEmpty("because original actor shouldn't receive result when delegated actor made the request");
+            peekResult = delegatedActorPeekResults.Should().ContainSingle("because there should only be one message for one grid area")
+                .Subject;
+
+            peekResult.Bundle.Should().NotBeNull("because peek result should contain a document stream");
+        }
+
+        await ThenNotifyWholesaleServicesDocumentIsCorrect(
+            peekResult.Bundle,
+            peekDocumentFormat,
+            new NotifyWholesaleServicesDocumentAssertionInput(
+                Timestamp: "2024-07-01T14:57:09Z",
+                BusinessReasonWithSettlementVersion: new BusinessReasonWithSettlementVersion(
+                    BusinessReason.WholesaleFixing,
+                    null),
+                ReceiverId: delegatedToActor.ActorNumber.Value,
+                ReceiverRole: originalActor.ActorRole,
+                SenderId: DataHubDetails.DataHubActorNumber.Value, // Sender is always DataHub
+                SenderRole: ActorRole.MeteredDataAdministrator,
+                ChargeTypeOwner: chargeOwnerNumber.Value,
+                ChargeCode: expectedChargeCode,
+                ChargeType: expectedChargeType,
+                Currency: exampleWholesaleResultMessageForActor.Currency,
+                EnergySupplierNumber: energySupplierNumber.Value,
+                SettlementMethod: exampleWholesaleResultMessageForActor.SettlementMethod,
+                MeteringPointType: exampleWholesaleResultMessageForActor.MeteringPointType,
+                GridArea: gridAreaCode,
+                transactionId,
+                PriceMeasurementUnit: exampleWholesaleResultMessageForActor.MeasurementUnit,
+                ProductCode: "5790001330590", // Example says "8716867000030", but document writes as "5790001330590"?
```

Comment:
```suggestion
                ProductCode: "5790001330590", // Example says "8716867000030", but document writes as "5790001330590"?
```
If the comment is copied. I think we should delete it